### PR TITLE
chore: added collection methods for module and collection service

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds retrieve and listAndCount to module service
- adds retrieve and listAndCount to collection service
- adds a generic retrieve helper for services

RESOLVES CORE-1428

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

File: packages/product/src/services/helpers/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from './retrieve-entity'
```
Comment (olivermrbl) [inline packages/product/src/services/helpers/index.ts]: todo: Double quotes

File: packages/product/src/services/helpers/retrieve-entity.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+type RetrieveEntityParams<TDTO> = {
+  id: string,
+  entityName: string,
+  repository: DAL.TreeRepositoryService
+  config: FindConfig<TDTO>
+  sharedContext?: Context
+}
+
+export async function retrieveEntity<
```
Comment (olivermrbl) [inline packages/product/src/services/helpers/retrieve-entity.ts]: praise: Nice, one could argue this should live in utils?

**Review by olivermrbl — APPROVED** [review]
LGTM - a minor todo before we merge

File: packages/types/src/product/common.ts [hunk 3]
```diff
@@ -96,6 +96,7 @@ export interface ProductCollectionDTO {
   handle: string
   metadata?: Record<string, unknown> | null
   deleted_at?: string | Date
+  products: ProductDTO[]
```
Comment (adrien2p) [inline packages/types/src/product/common.ts]: todo: optional I would say, it depends if we populate the relation right?

File: packages/product/src/services/helpers/retrieve-entity.ts (hunk 2 above)
Comment (adrien2p) [reply packages/product/src/services/helpers/retrieve-entity.ts]: agree, in utils/src/modules-sdk?

**Review by adrien2p — APPROVED** [review]
LGTM plus couple of todo

**Pull request merged**